src/flowMax.coffee

```diff
@@ -16,27 +16,34 @@ flowMax = (...funcs) ->
       funcs[0...index]
   getFollowingFuncs = (index) ->
     funcs[(index + 1)..]
+  flowLength = funcs?.length ? 0
+  if flowLength
+    for func, funcIndex in funcs
+      if (getNestedFlowMaxArguments = isFlowMax func)
+        return flowMax(
+          ...getPrecedingFuncs(funcIndex)
+          ...getNestedFlowMaxArguments()
+          ...getFollowingFuncs(funcIndex)
+        )
+      if isAddPropTypes(func) or isAddWrapper func
+        newFlowMax = flowMax(
+          ...getPrecedingFuncs(funcIndex)
+          func flowMax ...getFollowingFuncs(funcIndex)
+        )
+        # Expose original arguments if we're nested
+        newFlowMax[getArgumentsPropertyName] = -> funcs
```

Comment:
The only thing that was slightly tricky (which is probably why I moved it to runtime in #5) is that for nested `flowMax()`'s, by default the compile-time transformation would result in the nested list of functions no longer being visible (since at least in the current implementation, it returns a new `flowMax()` when it sees an `addPropTypes()`/`addWrapper()`, which would have a different list of functions exposed)

So just had to manually override the function-list-fetching hook on the new `flowMax()` to expose the original list of functions